src/get-started/flutter-for/ios-devs.md

```diff
@@ -5,467 +5,838 @@ description: Learn how to apply iOS developer knowledge when building Flutter ap
+to represent parts of your app's UI, 
+and you can provide modifiers that are used to configure those views. 
 
-To update your views on iOS, you directly mutate them.
-In Flutter, widgets are immutable and not updated directly.
-Instead, you have to manipulate the widget’s state.
+<img src="/assets/images/docs/get-started/ios/swiftui-modifiers.png" alt="SwiftUI Text view uses a padding modifier">
 
-This is where the concept of Stateful vs Stateless widgets
-comes in. A `StatelessWidget` is just what it sounds
-like&mdash;a widget with no state attached.
+Flutter is similar in that it uses **_widgets_** 
+to represent parts of your app's UI. Both views and widgets are immutable, 
+meaning they only exist until they need to be changed. 
+Additionally, both SwiftUI and Flutter are designed to use composition 
+rather than inheritance. However, Flutter uses individual widget classes 
+for many aspects of the UI. This means that properties 
+that might normally be represented as modifiers in SwiftUI 
+are instead represented as separate widgets in Flutter. 
 
-`StatelessWidgets` are useful when the part of the user interface you are
-describing does not depend on anything other than the initial configuration
-information in the widget.
+<img src="/assets/images/docs/get-started/ios/flutter-widgets.png" alt="Flutter Text Widget is encapsulated by a Padding widget">
 
-For example, in iOS, this is similar to placing a `UIImageView`
-with your logo as the `image`. If the logo is not changing during runtime,
-use a `StatelessWidget` in Flutter.
+In both SwiftUI and Flutter, 
+layouts are composed by nesting views or widgets, 
+respectively, within one another.
 
-If you want to dynamically change the UI based on data received
-after making an HTTP call, use a `StatefulWidget`.
-After the HTTP call has completed, tell the Flutter framework
-that the widget’s `State` is updated, so it can update the UI.
+### Layout process
 
-The important difference between stateless and
-stateful widgets is that `StatefulWidget`s have a `State` object
-that stores state data and carries it over across tree rebuilds,
-so it's not lost.
+In SwiftUI, the layout process is composed of a few different steps:
 
-If you are in doubt, remember this rule:
-if a widget changes outside of the `build` method
-(because of runtime user interactions, for example),
-it’s stateful.
-If the widget never changes, once built, it's stateless.
-However, even if a widget is stateful, the containing parent widget
-can still be stateless if it isn’t itself reacting to those changes
-(or other inputs).
+1. The parent view proposes some size to the child view.
+1. The child decides the size it requires by going through 
+this same process with its own list of children. 
+It passes down the proposed size and 
+asks its child what size it wishes to be. 
+It then returns the size it requires to its parent.
+1. The parent lays out the child, respecting the size that the child requested.
```

Comment:
This feels more confusing to me

Reply:
Yeah... It was somewhat to me, too. After I read the Flutter version of steps, I think this could have been written similarly.

The distinctions for SwiftUI, if I read this correctly, are:

1. The top-level parent proposes a size to its child view.
2. All subsequent child views also propose a size to their child's view while asking that child what size it _wants_ to be.
3. Each child view returns a size both desired and allowed to its parent.
4. Each parent view renders the its child view at the returned size.
5. Steps 2-4 repeat until the top-level parent renders its child.

Does that read more clearly?

I know that Flutter only offers limits to its children while asking those children what size they want.

This means the main difference is that SwiftUI proposes a size while Flutter provides constraints.